extensions/amp-analytics/0.1/transport.js

```diff
@@ -69,46 +78,57 @@ export class Transport {
 
   /**
    * @param {string} url
-   * @param {!Array<!batchSegmentDef>} segments
+   * @param {!Array<!BatchSegmentDef>} segments
+   * @param {boolean} inBatch
    */
-  sendRequest(url, segments) {
-    const serializer = this.getSerializer_();
-    const request = serializer(url, segments);
-    if (!request) {
-      user().error(TAG_, 'Request not sent. Contents empty.');
+  sendRequest(url, segments, inBatch) {
+    if (!url || segments.length === 0) {
+      dev().info(TAG_, 'Empty request not sent: ', url);
       return;
     }
+    const serializer = this.getSerializer_();
+    /**
+     * @param {boolean} withPayload
+     * @return {!RequestDef}
+     */
+    function generateRequest(withPayload) {
+      const request = inBatch
+        ? serializer.generateBatchRequest(url, segments, withPayload)
+        : serializer.generateRequest(url, segments[0], withPayload);
```

Comment:
I don't all the way understand the segments stuff, but is there ever a case where we don't want the first item in this array?

Reply:
in non-batch mode, every analytics trigger are sent immediately and the array is flushed. so it will always be the 1st item.

Reply:
Makes sense, thanks.